File: 2021/0013_romanToInt/0830/romanToInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        VALUES = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        ret = 0
        for i in range(len(s) - 1):
            value = VALUES[s[i]]
            if value < VALUES[s[i+1]]:
                ret -= value
            else:
                ret += value
        return ret + VALUES[s[-1]]
            

File: 2021/0013_romanToInt/0830/romanToInt.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        s.bytes().rev().fold((0, 0), |acc, cur| {
            let value = match cur {
                b'I' => 1,   b'V' => 5,   b'X' => 10,   b'L' => 50,
                b'C' => 100, b'D' => 500, b'M' => 1000, _ => -9999
            };
            (if value < acc.1 {acc.0 - value} else {acc.0 + value}, value)
        }).0
    }
}
